### PR TITLE
chore: use prettier on markdown files

### PR DESCRIPTION
**Pull request by ssbarnea**
This should ensure that we no longer get surprise changes when we edit tracked `.md` files in vscode as it applies all prettier rules, while still passing our markdown linting.

Please note that markdown files touched by this change are result of running pre-commit, no other manual changes were made.

Based on https://github.com/ansible/vscode-ansible/pull/386

### CODE REVIEW

File: .github/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -1,6 +1,5 @@
 # Community Code of Conduct
 
-Please see the official [Ansible Community Code of Conduct][CoC].
```
Comment (webknjaz) [inline .github/CODE_OF_CONDUCT.md:3]: This doesn't make sense to lowercase — it's a commonly used abbreviation.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -5,12 +5,12 @@ In order to contribute, you'll need to:
 1. Fork the repository.
 
 2. Create a branch, push your changes there. Don't forget to
-   {ref}`include news files for the changelog <_ansible_language_server_adding_changelog_fragments>`.
```
Comment (webknjaz) [inline .github/CONTRIBUTING.md:8]: Already commented in the other pr.

File: .github/SECURITY.md [hunk 3]
```diff
@@ -2,15 +2,13 @@
...
-vulnerabilities. Please read our [policies for reporting bugs][bug
-reports policy] if you want to report a security issue that might affect
-Ansible.
+We encourage responsible disclosure practices for security vulnerabilities.
+Please read our [policies for reporting bugs][bug reports policy] if you want to
+report a security issue that might affect Ansible.
 
-[bug reports policy]:
-https://docs.ansible.com/ansible/devel/community/reporting_bugs_and_features.html#reporting-a-bug
+[bug reports policy]: https://docs.ansible.com/ansible/devel/community/reporting_bugs_and_features.html#reporting-a-bug
```
Comment (webknjaz) [inline .github/SECURITY.md:14]: Don't do this, it's too long and should live on a separate line.

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -66,14 +66,20 @@ repos:
...
+        types_or: ["markdown"]
   - repo: https://github.com/igorshubovych/markdownlint-cli
     rev: v0.30.0
     hooks:
-    - id: markdownlint
-      exclude: >
-        (?x)^
-          docs/changelog-fragments\.d/\d+\.\w+\.md
-        $
+      - id: markdownlint
```
Comment (webknjaz) [inline .pre-commit-config.yaml:78]: Unrelated.

File: .prettierrc.yaml [hunk 5]
```diff
@@ -2,9 +2,14 @@ tabWidth: 2
 useTabs: false
 overrides:
   - files:
-      - '*.yaml'
-      - '*.yml'
+      - "*.yaml"
```
Comment (webknjaz) [inline .prettierrc.yaml:5]: Unrelated.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,9 +1,11 @@
 <!-- markdownlint-disable no-duplicate-heading no-multiple-blanks -->
+
 # Change Log
 
-All notable changes to the Ansible VS Code extension will be documented in this file.
+All notable changes to the Ansible VS Code extension will be documented in this
+file.
 
-[//]: # (DO-NOT-REMOVE-versioning-promise-START)
```
Comment (webknjaz) [inline CHANGELOG.md:6]: Commented on the other pr.

File: README.md [hunk 7]
```diff
@@ -59,23 +59,22 @@ The extension tries to detect whether the cursor is on a play, block or task
...
+- the `name` property is always suggested first
+- on module options, the required properties are shown first, and aliases are
   shown last, otherwise ordering from the documentation is preserved
-* FQCNs (fully qualified collection names) are inserted only when necessary;
+- FQCNs (fully qualified collection names) are inserted only when necessary;
   collections configured with the [`collections` keyword] are honored. This
   behavior can be disabled in extension settings.
 
-[`collections` keyword]:
-https://docs.ansible.com/ansible/latest/user_guide/collections_using.html#simplifying-module-names-with-the-collections-keyword
```
Comment (webknjaz) [inline README.md:70]: Same: long lines should be multilined in such places.

File: README.md [hunk 8]
```diff
@@ -135,30 +133,30 @@ The following settings are supported.
 For details on setting up development environment and debugging refer to the
 [development document].
 
-[development document]:
-https://github.com/ansible/ansible-language-server/blob/main/docs/development.md
```
Comment (webknjaz) [inline README.md:139]: Same: don't produce long lines here.

File: docs/changelog-fragments.d/158.misc.md [hunk 9]
```diff
@@ -1,4 +1,4 @@
-Added changelog fragment management infrastructure using [Towncrier]
--- by {user}`webknjaz`
+Added changelog fragment management infrastructure using [Towncrier] -- by
+{user}`webknjaz`
 
-[Towncrier]: https://github.com/twisted/towncrier
+[towncrier]: https://github.com/twisted/towncrier
```
Comment (webknjaz) [inline docs/changelog-fragments.d/158.misc.md:4]: No need to change the spelling of a tool.

File: docs/changelog-fragments.d/161.misc.md [hunk 10]
```diff
@@ -1,4 +1,4 @@
-Added [Sphinx] documentation generator and set up the CI
-infrastructure for it -- by {user}`webknjaz`
+Added [Sphinx] documentation generator and set up the CI infrastructure for it
+-- by {user}`webknjaz`
 
-[Sphinx]: https://github.com/twisted/towncrier
+[sphinx]: https://github.com/twisted/towncrier
```
Comment (webknjaz) [inline docs/changelog-fragments.d/161.misc.md:4]: Don't break the spelling of the tool.

File: docs/changelog-fragments.d/164.doc.md [hunk 11]
```diff
@@ -1,5 +1,5 @@
-Dropped the brackets from the changelog titles for the release sections.
-We now don't strictly follow the release notes format suggested by [Keep
-a Changelog] -- by {user}`webknjaz`
+Dropped the brackets from the changelog titles for the release sections. We now
+don't strictly follow the release notes format suggested by [Keep a Changelog]
+-- by {user}`webknjaz`
 
-[Keep a Changelog]: https://keepachangelog.com/en/1.1.0/
+[keep a changelog]: https://keepachangelog.com/en/1.1.0/
```
Comment (webknjaz) [inline docs/changelog-fragments.d/164.doc.md:5]: This is particularly weird — it's hard to search for non-matching names because of the differing case.

File: docs/changelog-fragments.d/README.md [hunk 12]
```diff
@@ -1,50 +1,43 @@
 <!-- markdownlint-disable first-line-heading -->
 
-(_ansible_language_server_adding_changelog_fragments)=
```
Comment (webknjaz) [inline docs/changelog-fragments.d/README.md:3]: I don't see how changing the anchors could be related.

File: docs/changelog-fragments.d/README.md [hunk 13]
```diff
@@ -57,23 +50,20 @@ Added a `{user}` role to Sphinx config -- by {user}`webknjaz`
 File `docs/changelog-fragments.d/116.feature.md`:
 
 ```md
-Added support for nested module options (suboptions)
--- by {user}`tomaciazek`
```
Comment (webknjaz) [inline docs/changelog-fragments.d/README.md:61]: This change doesn't make sense, it was demonstrating that it is okay to have more than one line in fragments.

File: docs/changelog-fragments.d/README.md [hunk 14]
```diff
@@ -57,23 +50,20 @@ Added a `{user}` role to Sphinx config -- by {user}`webknjaz`
...
 See `pyproject.toml` for all available categories
 (`tool.towncrier.type`).
 ```
 
-[MyST Markdown]:
-https://myst-parser.rtfd.io/en/latest/syntax/syntax.html
-[Towncrier philosophy]:
-https://towncrier.rtfd.io/en/actual-freaking-docs/#philosophy
+[myst markdown]: https://myst-parser.rtfd.io/en/latest/syntax/syntax.html
+[towncrier philosophy]: https://towncrier.rtfd.io/en/actual-freaking-docs/#philosophy
```
Comment (webknjaz) [inline docs/changelog-fragments.d/README.md:69]: Same problem with breaking the case.

File: docs/changelog-fragments.d/README.md [hunk 15]
```diff
@@ -1,50 +1,43 @@
 <!-- markdownlint-disable first-line-heading -->
 
-(_ansible_language_server_adding_changelog_fragments)=
+(ansible_language_server_adding_changelog_fragments)=
 
 ## Adding change notes with your PRs
 
-It is very important to maintain a log for news of how
```
Comment (webknjaz) [inline docs/changelog-fragments.d/README.md:7]: This huge rewrite contributes to the complexity of synchronizing this text across projects 👎

File: docs/contributing/guidelines.md [hunk 16]
```diff
@@ -38,21 +40,19 @@ _______________________________________________________ summary ________________
...
+integrated [MyST parser] allowing us to also accept new documents written in an
+extended version of Markdown that supports using Sphinx directives and roles.
+{ref}`Read the docs <myst:intro/writing>` to learn more on how to use it.
 
-[MyST parser]: https://pypi.org/project/myst-parser/
-[Read The Docs]: https://readthedocs.org
-[Sphinx]: https://www.sphinx-doc.org
+[myst parser]: https://pypi.org/project/myst-parser/
+[read the docs]: https://readthedocs.org
+[sphinx]: https://www.sphinx-doc.org
```
Comment (webknjaz) [inline docs/contributing/guidelines.md:54]: Same with breaking the case.

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: .github/CODE_OF_CONDUCT.md (hunk 1 above)
Comment (ssbarnea) [reply .github/CODE_OF_CONDUCT.md:3]: That is not user visible. Prettier seems to enforce use of lowercase reference keys and is not configurable. I do not see any problem with that. I do not this that either as actionable or fixable.

Comment (webknjaz) [reply .github/CODE_OF_CONDUCT.md:3]: It's a big problem because it adds friction to cross repo sync + breaks expectations of identifiers being searchable with a standard case-sensitive search.

File: .github/SECURITY.md [hunk 17]
```diff
@@ -2,15 +2,14 @@
...
-vulnerabilities. Please read our [policies for reporting bugs][bug
-reports policy] if you want to report a security issue that might affect
-Ansible.
+We encourage responsible disclosure practices for security vulnerabilities.
+Please read our [policies for reporting bugs][bug reports policy] if you want to
+report a security issue that might affect Ansible.
 
 [bug reports policy]:
-https://docs.ansible.com/ansible/devel/community/reporting_bugs_and_features.html#reporting-a-bug
+  https://docs.ansible.com/ansible/devel/community/reporting_bugs_and_features.html#reporting-a-bug
```
Comment (webknjaz) [inline .github/SECURITY.md:15]: Why indent this?

Comment (ssbarnea) [reply .github/SECURITY.md:15]: In fact I do remember a rule in markdownlint about splitting long lines. I am surprised that markdown-lint did not complain about that line because that wrapping rule would have requested the wrapping.

That is happening only because the URL is very long, not always.

**Review by webknjaz — CHANGES_REQUESTED** [review]
I'm still against lowecasing. Please respect the previous agreement of not marking conversations as resolved because the requests there are not resolved.

File: docs/changelog-fragments.d/README.md (hunk 13 above)
Comment (ssbarnea) [reply docs/changelog-fragments.d/README.md:62]: That is clearly an unbacked statement as there is no need need to wrap text to show people they can wrap it or not. We are not in the business to teach people markdown syntax.

If you really wanted to demonstrate it you would have included something like `This can also wrap on multiple lines." as part of the fragment itself.

As we would introduce automatic formatting there we have no need to tell user how and where to wrap the text as that will be done automatically.

File: docs/changelog-fragments.d/README.md (hunk 15 above)
Comment (ssbarnea) [reply docs/changelog-fragments.d/README.md:7]: @ansible/devtools team never agreed on something like this. We do not have a reference sync source and the reality is that even if we would do, we could implement the same formatting on the other one.

Basically, i see the comment as diverging from the scope of the PR and no real reasons for excluding this file, especially as it will render any attempt to touch it with vscode a nightmare as prettier will reformat it.

**Comment by ssbarnea** [comment]
> I'm still against lowecasing. Please respect the previous agreement of not marking conversations as resolved because the requests there are not resolved.

That floating comment makes no sense because there is no lowercase change as part of this PR. The lowercasing was removed from this PR and that is why the original comment was marked as resolved, as it was not longer applying to that change.

**Comment by webknjaz** [comment]
> That floating comment makes no sense because there is no lowercase change as part of this PR.

It was here when I asked not to do it, and it was still implemented by sneakily merging it despite the request. If there is no value in giving reviews and people merge things via by separating patches and hiding them from reviewers who explicitly asked not to do them in the first place, is that our new accepted way of collaboration? Do you expect me to use this new best practice as well as a primary workaround?

**Comment by webknjaz** [comment]
> that is why the original comment was marked as resolved

It was not resolved, you did exactly opposite to what was asked. Is this what resolved actually means now?
